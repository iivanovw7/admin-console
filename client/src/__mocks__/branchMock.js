export const BRANCH = {
  address: '938 Barby Way',
  email: 'mbrazier0@amazon.co.uk',
  fax: '+86 829 425 7497',
  information: '1',
  name: 'San Mart√≠n de los Andes',
  phone: '+54 275 474 9318',
  status: true,
  __v: 0,
  _id: '5c8f96b9cece9911f01dd092'
};