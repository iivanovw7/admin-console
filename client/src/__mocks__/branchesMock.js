export const BRANCHES = {
  page: 1,
  limit: 7,
  pages: 4,
  results: 25,
  output: [
    {
      address: '938 Barby Way',
      email: 'mbrazier0@amazon.co.uk',
      fax: '+86 829 425 7497',
      information: '1',
      name: 'San Mart√≠n de los Andes',
      phone: '+54 275 474 9318',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd092'
    },
    {
      address: '903 Dottie Street',
      email: 'inerger1@businesswire.com',
      fax: '+86 230 609 6364',
      information: 't',
      name: 'Dingzhai',
      phone: '+86 103 802 9655',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd093'
    },
    {
      status:false,
      _id:"5c8f96b9cece9911f01dd0a0",
      name:"Cantao-an",
      email:"atroye@dailymail.co.uk",
      phone:"+63 269 360 0515",
      fax:"+86 713 306 1827",
      address:"7827 Luster Point",
    }
  ]
};
