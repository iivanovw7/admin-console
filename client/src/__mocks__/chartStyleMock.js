export const style = {
  type: "PieChart",
  id: 2
};
