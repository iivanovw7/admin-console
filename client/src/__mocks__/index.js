export * from './branchesMock';
export * from './branchMock';
export * from './userMock';
export * from './usersMock';
export * from './groupMock';
export * from './groupsMock';
export * from './roleMock';
export * from './rolesMock';
export * from './ticketMock';
export * from './ticketsMock';
export * from './messageMocks';
export * from './messagesMocks';
export * from './statisticsMock';
