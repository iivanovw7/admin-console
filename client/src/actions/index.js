export * from './authActions';
export * from './branchesActions';
export * from './groupsActions';
export * from './rolesActions';
export * from './usersActions';
export * from './ticketsActions';
export * from './messagesActions';
export * from './statisticsActions';