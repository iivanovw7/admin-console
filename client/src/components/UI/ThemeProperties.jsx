import blue from '@material-ui/core/colors/blue';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { createMuiTheme } from '@material-ui/core/styles';

const drawerWidth = 240;

//Exporting material-ui main themeProperties configuration
export const themeProperties = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  mixins: {
    toolbar: {
      textAlign: 'center',
      minHeight: '64px',
      justifyContent: 'space-between'
    }
  },
  palette: {
    primary: blue,
    secondary: {
      main: '#ffffff'
    }
  },
  status: {
    danger: 'orange'
  }
});

//Exporting material-ui main styles
export const NavigationStyles = theme => ({
  active: {
    backgroundColor: 'rgba(0, 0, 0, 0.10)'
  },
  root: {
    display: 'flex'
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0
    }
  },
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth}px)`
    }
  },
  menuButton: {
    marginRight: 20,
    [theme.breakpoints.up('sm')]: {
      display: 'none'
    }
  },
  toolbar: theme.mixins.toolbar,
  toolbarUserName: {
    [theme.breakpoints.down('xs')]: {
      display: 'none'
    }
  },
  drawerPaper: {
    width: drawerWidth
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    marginRight: theme.spacing.unit * 2,
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit * 3,
      width: 'auto'
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  inputRoot: {
    color: 'inherit',
    width: '100%'
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: 200
    }
  }
});

export const LoginFormStyles = theme => ({

  [theme.breakpoints.down('sm')]: {
    loginFormContainer: {
      boxShadow: 'none',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center'
    }
  },
  loginWrapper: {
    maxWidth: '500px',
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%'
  },
  loginFormContainer: {
    padding: theme.spacing.unit,
    maxWidth: '500px'
  },
  formPaper: {
    margin: theme.spacing.unit * 2
  },
  paddingTop: theme.spacing.unit * 2,
  paddingBottom: theme.spacing.unit * 2

});

//Exporting material-ui main styles
export const Wrapper = theme => ({
  root: {
    display: 'flex'
  },
  toolbar: theme.mixins.toolbar,
  contentSingle: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  contentList: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    [theme.breakpoints.up('sm')]: {
      height: '90vh'
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    }
  },
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap',

    buttonWrapper: {
      display: 'flex',
      alignItems: 'center'
    }
  },
  fromLimitSelector: {
    display: 'flex',
    flexWrap: 'wrap',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
    maxWidth: 300
  },
  controlsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingRight: 10,
    paddingLeft: 10
  },
  selectorsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: 10,
    paddingLeft: 10
  }
});

export const Container = theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
    display: 'flex',
    justifyContent: 'center',
    overflow: 'hidden'
  },
  tables: {
    minWidth: 200,
    width: '100%'
  },
  tableCell: {
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5
  },
  rowClass: {
    '&:hover': {
      backgroundColor: fade(theme.palette.common.black, 0.1),
      cursor: 'pointer'
    },
    [theme.breakpoints.up('sm')]: {
      height: '48px',
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    },
  },
  nameCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,

    [theme.breakpoints.up('sm')]: {
      width: '30%',
    },

    [theme.breakpoints.down('sm')]: {
      width: '60%',
    }

  },
  addressCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '100%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  employeesCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '30%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  branchPaper: {
    width: '100%',
    maxWidth: 600,
    margin: theme.spacing.unit * 2
  }
});

export const Notification = theme => ({
  close: {
    padding: theme.spacing.unit / 2
  }
});

export const Buttons = theme => ({
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap'
  },
  [theme.breakpoints.down('sm')]: {
    mobile: {
      display: 'flex'
    },
    desktop: {
      display: 'none'
    }
  },
  [theme.breakpoints.up('sm')]: {
    mobile: {
      display: 'none'
    },
    desktop: {
      display: 'flex'
    }
  }
});

