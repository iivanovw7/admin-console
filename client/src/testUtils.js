import checkPropTypes from 'check-prop-types';
import { createStore } from 'redux';
import rootReducer from './reducers';

/**
 * @param {object} initialState - initial state to store
 * @function storeFactory
 * @returns {Store} Redux store
 */
export const storeFactory = initialState => {
  return createStore(rootReducer, initialState);
};

/**
 * Returns node(s) with the given test attributes
 * @param {object} wrapper - shallow wrapper
 * @param {string} val - data-test attribute to find
 * @returns {wrapper}
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test='${val}']`);
};