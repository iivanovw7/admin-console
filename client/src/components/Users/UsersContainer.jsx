import { Link, Paper, Table, TableBody, TableCell, TableHead, TableRow } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import {
  getBranches,
  getGroups,
  getRoles,
  getSingleUser,
  getUsers,
  searchUsers
} from '../../actions/index';
import AlertSnackbar from '../UI/Notifications/Snackbar';
import { Container } from '../UI/ThemeProperties';
import { actionCheckBox } from './UsersActionCheckBox';

const UsersContainer = props => {
  const { classes, history, dispatch, users, limit, page, search } = props;
  const currPage = page || 1;
  const currLimit = limit || 8;

  function displayStatus(status) {
    return status ? 'Active' : 'Disabled';
  }

  function handleUserClick(id) {
    dispatch(getSingleUser(id, history));
    dispatch(getRoles(1, 100, history));
    dispatch(getGroups(1, 100, history));
    dispatch(getBranches(1, 100, history));
  }

  //Triggers notification if there are any messages in props
  function displayNotifications() {
    if (props.errorMessage && !props.messageConfirmed) {
      return displayNotification(props.errorMessage, false);
    }
    if (props.successMessage && !props.messageConfirmed) {
      return displayNotification(props.successMessage, true);
    }
  }

  //Returns notification container
  const displayNotification = (message, success) => (

    <AlertSnackbar
      dispatch={dispatch}
      message={message}
      afterConfirm={() => {
        search ?
          dispatch(searchUsers(currPage, currLimit, search, history)) :
          dispatch(getUsers(currPage, currLimit, history));
      }}
      success={success}
    />
  );

  return (
    <Paper className={classes.root}>
      <Table className={classes.tables}>
        <TableHead>
          <TableRow>
            <TableCell className={classes.userNameCell}>
              <h3>Name</h3>
            </TableCell>
            <TableCell className={classes.userEmailCell} align="center">
              <h3>Email</h3>
            </TableCell>
            <TableCell className={classes.userStatusCell} align="center">
              <h3>Status</h3>
            </TableCell>
            <TableCell className={classes.userControlCell} align="center">
              <h3>Deactivate</h3>
            </TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {users.map(row => (
            <TableRow key={row._id}>
              <TableCell component="th" scope="row" className={classes.userNameCell}>
                <Link
                  component="button"
                  variant="body2"
                  onClick={() => {
                    handleUserClick(row._id, history);
                  }}
                >
                  {row.name} {row.surname}
                </Link>
              </TableCell>
              <TableCell className={classes.userEmailCell} align="center">
                {row.email}
              </TableCell>
              <TableCell className={classes.userStatusCell} align="center">
                <strong>
                  {displayStatus(row.status)}
                </strong>
              </TableCell>
              <TableCell className={classes.userControlCell} align="center">
                {actionCheckBox(row, classes, dispatch, history)}
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
      {displayNotifications()}
    </Paper>
  );
};

UsersContainer.propTypes = {
  classes: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    errorMessage: state.users.error,
    successMessage: state.users.success,
    messageConfirmed: state.users.confirmed
  };
}

export default connect(mapStateToProps, {
  getUsers,
  getSingleUser
})(withStyles(Container)(withRouter(UsersContainer)));