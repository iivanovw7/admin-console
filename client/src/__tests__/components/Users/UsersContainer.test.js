import { withStyles } from '@material-ui/core';
import { createMount, createShallow } from '@material-ui/core/test-utils';
import { createMemoryHistory } from 'history';
import jest from 'jest-mock';
import React from 'react';
import { connect, Provider } from 'react-redux';
import { MemoryRouter, withRouter } from 'react-router-dom';
import { getUsers } from '../../../actions';
import { Container } from '../../../components/UI/ThemeProperties';
import UsersContainer from '../../../components/Users/UsersContainer';
import { storeFactory } from '../../../testUtils';
import * as mocks from './../../../__mocks__/';

const setup = (initialState = {}) => {
  let wrapper;
  let shallowWithDive;
  let mount;
  let shallow;
  let store = storeFactory(initialState);
  const history = createMemoryHistory('/dashboard');

  const Composition = connect(null, { getUsers })(withStyles(Container, { withTheme: true })(withRouter(UsersContainer)));
  const props = {
    limit: 10,
    page: 2,
    search: mocks.USER.name,
    history: history,
    users: mocks.USERS.output
  };

  beforeEach(() => {
    shallow = createShallow();
    shallowWithDive = createShallow({ dive: true });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  /**
   *  Renders users container with styles,
   *  Checks if component rendered properly.
   */
  describe('Users container', () => {
    it('Should render HOC with styles properly', () => {
      const mockFunc = jest.fn();
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
            <Composition {...props} dispatch={mockFunc}/>
          </MemoryRouter>
        </Provider>
      );
      expect(wrapper.html()).not.toBe(null);
      expect(wrapper.length).toEqual(1);
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
};

setup();



