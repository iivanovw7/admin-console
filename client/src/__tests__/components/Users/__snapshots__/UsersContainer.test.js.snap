// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users container Should render HOC with styles properly 1`] = `"<div class=\\"MuiPaper-root-53 MuiPaper-elevation2-57 MuiPaper-rounded-54 withRouter-UsersContainer--root-27 withRouter-Connect-WithStyles-withRouter-UsersContainer-----root-1\\"><table class=\\"MuiTable-root-80 withRouter-UsersContainer--tables-33 withRouter-Connect-WithStyles-withRouter-UsersContainer-----tables-7\\"><thead class=\\"MuiTableHead-root-81\\"><tr class=\\"MuiTableRow-root-82 MuiTableRow-head-85\\"><th class=\\"MuiTableCell-root-87 MuiTableCell-head-88 withRouter-UsersContainer--userNameCell-45 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userNameCell-19\\" scope=\\"col\\"><h3>Name</h3></th><th class=\\"MuiTableCell-root-87 MuiTableCell-head-88 MuiTableCell-alignCenter-96 withRouter-UsersContainer--userEmailCell-46 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userEmailCell-20\\" scope=\\"col\\"><h3>Email</h3></th><th class=\\"MuiTableCell-root-87 MuiTableCell-head-88 MuiTableCell-alignCenter-96 withRouter-UsersContainer--userStatusCell-47 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userStatusCell-21\\" scope=\\"col\\"><h3>Status</h3></th><th class=\\"MuiTableCell-root-87 MuiTableCell-head-88 MuiTableCell-alignCenter-96 withRouter-UsersContainer--userControlCell-48 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userControlCell-22\\" scope=\\"col\\"><h3>Deactivate</h3></th></tr></thead><tbody class=\\"MuiTableBody-root-99\\"><tr class=\\"MuiTableRow-root-82\\"><th class=\\"MuiTableCell-root-87 MuiTableCell-body-89 withRouter-UsersContainer--userNameCell-45 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userNameCell-19\\" scope=\\"row\\"><button class=\\"MuiTypography-root-105 MuiTypography-body2-113 MuiTypography-colorPrimary-135 MuiTypography-inline-140 MuiLink-root-100 MuiLink-button-104 MuiLink-underlineHover-102\\">Admin ADMIN</button></th><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-UsersContainer--userEmailCell-46 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userEmailCell-20\\">admin@company.org</td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-UsersContainer--userStatusCell-47 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userStatusCell-21\\"><strong>Active</strong></td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-UsersContainer--userControlCell-48 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userControlCell-22\\"><span class=\\"MuiButtonBase-root-157 MuiIconButton-root-151 MuiPrivateSwitchBase-root-147 MuiCheckbox-root-141 MuiCheckbox-colorPrimary-145 MuiPrivateSwitchBase-checked-148 MuiCheckbox-checked-142\\" style=\\"height: 16px;\\"><span class=\\"MuiIconButton-label-156\\"><svg class=\\"MuiSvgIcon-root-160\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\\"></path></svg><input class=\\"MuiPrivateSwitchBase-input-150\\" type=\\"checkbox\\" data-indeterminate=\\"false\\" value=\\"\\" checked=\\"\\"></span><span class=\\"MuiTouchRipple-root-169\\"></span></span></td></tr><tr class=\\"MuiTableRow-root-82\\"><th class=\\"MuiTableCell-root-87 MuiTableCell-body-89 withRouter-UsersContainer--userNameCell-45 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userNameCell-19\\" scope=\\"row\\"><button class=\\"MuiTypography-root-105 MuiTypography-body2-113 MuiTypography-colorPrimary-135 MuiTypography-inline-140 MuiLink-root-100 MuiLink-button-104 MuiLink-underlineHover-102\\">Holt Aasaf</button></th><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-UsersContainer--userEmailCell-46 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userEmailCell-20\\">haasaf4q@tamu.edu</td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-UsersContainer--userStatusCell-47 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userStatusCell-21\\"><strong>Active</strong></td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-UsersContainer--userControlCell-48 withRouter-Connect-WithStyles-withRouter-UsersContainer-----userControlCell-22\\"><span class=\\"MuiButtonBase-root-157 MuiIconButton-root-151 MuiPrivateSwitchBase-root-147 MuiCheckbox-root-141 MuiCheckbox-colorPrimary-145 MuiPrivateSwitchBase-checked-148 MuiCheckbox-checked-142\\" style=\\"height: 16px;\\"><span class=\\"MuiIconButton-label-156\\"><svg class=\\"MuiSvgIcon-root-160\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\\"></path></svg><input class=\\"MuiPrivateSwitchBase-input-150\\" type=\\"checkbox\\" data-indeterminate=\\"false\\" value=\\"\\" checked=\\"\\"></span><span class=\\"MuiTouchRipple-root-169\\"></span></span></td></tr></tbody></table></div>"`;
